### PR TITLE
[Outlook] (sample) Including OWA handling for getSelectedDataAsync

### PR DESCRIPTION
**Pull request by ElizabethSamuel-MSFT**

### CODE REVIEW

**Comment by o365devx** [comment]
Docs Build status updates of commit _[60ae405](https://github.com/OfficeDev/office-js-docs-pr/commits/60ae405a41c4e277e6f0ab0364df3c7e69c7c39d)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/reference/objectmodel/preview-requirement-set/office.context.mailbox.item.md](https://github.com/OfficeDev/office-js-docs-pr/blob/elizs-getselected-example/docs/reference/objectmodel/preview-requirement-set/office.context.mailbox.item.md) | :white_check_mark:Succeeded | |
[docs/reference/objectmodel/requirement-set-1.2/office.context.mailbox.item.md](https://github.com/OfficeDev/office-js-docs-pr/blob/elizs-getselected-example/docs/reference/objectmodel/requirement-set-1.2/office.context.mailbox.item.md) | :white_check_mark:Succeeded | |
[docs/reference/objectmodel/requirement-set-1.3/office.context.mailbox.item.md](https://github.com/OfficeDev/office-js-docs-pr/blob/elizs-getselected-example/docs/reference/objectmodel/requirement-set-1.3/office.context.mailbox.item.md) | :white_check_mark:Succeeded | |
[docs/reference/objectmodel/requirement-set-1.4/office.context.mailbox.item.md](https://github.com/OfficeDev/office-js-docs-pr/blob/elizs-getselected-example/docs/reference/objectmodel/requirement-set-1.4/office.context.mailbox.item.md) | :white_check_mark:Succeeded | |
[docs/reference/objectmodel/requirement-set-1.5/office.context.mailbox.item.md](https://github.com/OfficeDev/office-js-docs-pr/blob/elizs-getselected-example/docs/reference/objectmodel/requirement-set-1.5/office.context.mailbox.item.md) | :white_check_mark:Succeeded | |
[docs/reference/objectmodel/requirement-set-1.6/office.context.mailbox.item.md](https://github.com/OfficeDev/office-js-docs-pr/blob/elizs-getselected-example/docs/reference/objectmodel/requirement-set-1.6/office.context.mailbox.item.md) | :white_check_mark:Succeeded | |
[docs/reference/objectmodel/requirement-set-1.7/office.context.mailbox.item.md](https://github.com/OfficeDev/office-js-docs-pr/blob/elizs-getselected-example/docs/reference/objectmodel/requirement-set-1.7/office.context.mailbox.item.md) | :white_check_mark:Succeeded | |
[docs/reference/objectmodel/requirement-set-1.8/office.context.mailbox.item.md](https://github.com/OfficeDev/office-js-docs-pr/blob/elizs-getselected-example/docs/reference/objectmodel/requirement-set-1.8/office.context.mailbox.item.md) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2019%5C11%5C6%5C57856ef7-e8cf-54ca-0d12-df5f42d1e8bb%5CPullRequest%5C201911062052322860-1422%5Cworkflow_report.html?sv=2016-05-31&sr=b&sig=sRx1st8%2Fu0SB5MgYXKARhcWKY5%2BkeqqEqBUgVW9WZEk%3D&st=2019-11-06T20%3A48%3A48Z&se=2019-12-07T20%3A53%3A48Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

File: docs/reference/objectmodel/preview-requirement-set/office.context.mailbox.item.md [hunk 1]
```diff
@@ -2342,11 +2340,13 @@ function getCallback(asyncResult) {
   var text = asyncResult.value.data;
   var prop = asyncResult.value.sourceProperty;
 
-  Office.context.mailbox.item.setSelectedDataAsync('Setting ' + prop + ': ' + text, {}, setCallback);
-}
+  // Handle where Outlook on the web erroneously returns "null" instead of empty string.
+  if (Office.context.mailbox.diagnostics.hostName === 'OutlookWebApp'
+      && asyncResult.value.endPosition === asyncResult.value.startPosition) {
```
Comment (AlexJerabek) [inline docs/reference/objectmodel/preview-requirement-set/office.context.mailbox.item.md]: This seems like a complicated way to check for null or an empty string. Is `value` a string?

File: docs/reference/objectmodel/preview-requirement-set/office.context.mailbox.item.md [hunk 2]
```diff
@@ -2302,9 +2302,7 @@ Asynchronously returns selected data from the subject or body of a message.
 If there is no selection but the cursor is in the body or subject, the method returns null for the selected data. If a field other than the body or subject is selected, the method returns the `InvalidSelection` error.
 
 > [!NOTE]
-> In Outlook on the web, the method returns the string "null" if no text is selected but the cursor is in the body. To check for this situation, include code similar to the following:
->
-> `var selectedText = (asyncResult.value.endPosition === asyncResult.value.startPosition) ? "" : asyncResult.value.data;`
+> In Outlook on the web, the method returns the string "null" if no text is selected but the cursor is in the body. To check for this situation, see the example later in this section.
```
[the remark this answers is not in the document]
Comment (ElizabethSamuel-MSFT) [reply docs/reference/objectmodel/preview-requirement-set/office.context.mailbox.item.md]: The actual string "null" (see GitHub issue https://github.com/OfficeDev/office-js/issues/800 for more details).

File: docs/reference/objectmodel/preview-requirement-set/office.context.mailbox.item.md (hunk 1 above)
Comment (ElizabethSamuel-MSFT) [reply docs/reference/objectmodel/preview-requirement-set/office.context.mailbox.item.md]: I'm not checking for a null or empty string. Instead, here we're checking that nothing has been selected so return an empty string.

File: docs/reference/objectmodel/preview-requirement-set/office.context.mailbox.item.md [hunk 3]
```diff
@@ -2302,9 +2302,7 @@ Asynchronously returns selected data from the subject or body of a message.
 If there is no selection but the cursor is in the body or subject, the method returns null for the selected data. If a field other than the body or subject is selected, the method returns the `InvalidSelection` error.
```
Comment (ElizabethSamuel-MSFT) [inline docs/reference/objectmodel/preview-requirement-set/office.context.mailbox.item.md]:
```suggestion
If there is no selection but the cursor is in the body or subject, the method returns an empty string for the selected data. If a field other than the body or subject is selected, the method returns the `InvalidSelection` error.
```

**Pull request merged**